packages/tailwindcss/src/utilities.ts

```diff
@@ -4318,20 +4321,162 @@ export function createUtilities(theme: Theme) {
       ['--tw-drop-shadow', ' '],
       ['filter', cssFilterValue],
     ])
-    functionalUtility('drop-shadow', {
-      themeKeys: ['--drop-shadow'],
-      handle: (value) => [
-        filterProperties(),
-        decl(
-          '--tw-drop-shadow',
-          segment(value, ',')
-            .map((v) => `drop-shadow(${v})`)
-            .join(' '),
-        ),
-        decl('filter', cssFilterValue),
-      ],
+
+    utilities.functional('drop-shadow', (candidate) => {
+      let alpha: string | undefined
+
+      if (candidate.modifier) {
+        if (candidate.modifier.kind === 'arbitrary') {
+          alpha = candidate.modifier.value
+        } else {
+          if (isPositiveInteger(candidate.modifier.value)) {
+            alpha = `${candidate.modifier.value}%`
+          }
+        }
+      }
+
+      if (!candidate.value) {
+        let value = theme.get(['--drop-shadow'])
+        if (value === null) return
+
+        return [
+          filterProperties(),
+          decl('--tw-drop-shadow-alpha', alpha),
+          decl(
+            '--tw-drop-shadow-size',
+            segment(value, ',')
+              .map(
+                (v) =>
+                  `drop-shadow(${replaceShadowColors(
+                    v,
+                    alpha,
+                    (color) => `var(--tw-drop-shadow-color, ${color})`,
+                  )})`,
+              )
+              .join(' '),
+          ),
+          decl('--tw-drop-shadow', `drop-shadow(${theme.resolve(null, ['--drop-shadow'])})`),
+          decl('filter', cssFilterValue),
+        ]
+      }
+
+      if (candidate.value.kind === 'arbitrary') {
+        let value: string | null = candidate.value.value
+        let type = candidate.value.dataType ?? inferDataType(value, ['color'])
+
+        switch (type) {
+          case 'color': {
+            value = asColor(value, candidate.modifier, theme)
+            if (value === null) return
+            return [
+              filterProperties(),
+              decl('--tw-drop-shadow-color', withAlpha(value, 'var(--tw-drop-shadow-alpha)')),
+              decl('--tw-drop-shadow', `var(--tw-drop-shadow-size)`),
+            ]
+          }
+          default: {
+            if (candidate.modifier && !alpha) return
+
+            return [
+              filterProperties(),
+              decl('--tw-drop-shadow-alpha', alpha),
+              decl(
+                '--tw-drop-shadow-size',
+                segment(value, ',')
+                  .map(
+                    (v) =>
+                      `drop-shadow(${replaceShadowColors(
+                        v,
+                        alpha,
+                        (color) => `var(--tw-drop-shadow-color, ${color})`,
+                      )})`,
```

Comment:
I think the API changed here because this is passing 3 values instead of 2

Reply:
Ah I think it's because I pressed the update button (because of conflicts in the CHANGELOG.md file)